### PR TITLE
feat: Add sha256 in CI for C builds - WPB-11313

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11313" title="WPB-11313" target="_blank">TaskWPB-11313</a>  Add sha256 in iOS CI especially for C builds
  </td></table>
  <br />
 

### Issue

We need to provide sha256 for C builds. This PR adds it to the build notification.

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -575,9 +575,9 @@ platform :ios do
             app_name = build.app_center_app_name
         end
 
-        
-        sh "echo SHA of IPA is:"
-        sh "shasum -a 256 ../#{build.artifact_path(with_filename: true)}.ipa || true"
+        sh "SHA256=$(shasum -a 256 ../#{build.artifact_path(with_filename: true)}.ipa || true)"
```
Comment (netbe) [inline fastlane/Fastfile]: are you sure it's the device ipa that we have here and not the simulator one?

Comment (KaterinaWire) [reply fastlane/Fastfile]: this is an artifact that we use to upload to appcenter. 
There are 2 approaches:
- get sha256 from this artifact
- or wait a result from appcenter, download ipa, and compute the sha256.
The first option sounds more efficient.

**Review by netbe — APPROVED** [review]
LGTM;)

**Review by caldrian — APPROVED** [review]
Is this not relevant for 3.115?

File: fastlane/Fastfile (hunk 1 above)
Comment (netbe) [reply fastlane/Fastfile]: sounds good! I guess I got confused as in upload_s3 we upload 2 ipas in some cases one for simulator (QA) and one for device, so i wasn't sure it was the right one here

**Comment by netbe** [comment]
> Is this not relevant for 3.115?

@KaterinaWire this should be added to 3.112.0 branch, as it's where we'll need it for C builds

**Pull request merged**